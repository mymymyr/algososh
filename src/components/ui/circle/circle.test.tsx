import { Circle } from './circle'
import renderer from 'react-test-renderer';
import { createElement } from "react";
import { ElementStates } from "../../../types/element-states";

describe('Тестирует компонент сircle', () => {
    it('Без буквы', () => {
        const component = renderer.create(<Circle />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С буквами', () => {
        const component = renderer.create(<Circle letter='test' />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С head', () => {
        const component = renderer.create(<Circle head='testHead' />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С react-элементом в head', () => {
        const reactElement = createElement("div");
        const component = renderer.create(<Circle head={reactElement} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С tail', () => {
        const component = renderer.create(<Circle tail='testTail' />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С react-элементом в tail', () => {
        const reactElement = createElement("div");
        const component = renderer.create(<Circle tail={reactElement} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С index', () => {
        const component = renderer.create(<Circle index={0} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('С пропом isSmall === true', () => {
        const component = renderer.create(<Circle isSmall />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('В состоянии default', () => {
        const component = renderer.create(<Circle state={ElementStates.Default} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('В состоянии changing', () => {
        const component = renderer.create(<Circle state={ElementStates.Changing} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('В состоянии modified', () => {
        const component = renderer.create(<Circle state={ElementStates.Modified} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
})
