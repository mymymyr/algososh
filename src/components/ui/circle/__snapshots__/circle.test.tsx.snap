// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирует компонент сircle Без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle В состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle В состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle В состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle С head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    testHead
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle С index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирует компонент сircle С react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div />
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle С react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div />
  </div>
</div>
`;

exports[`Тестирует компонент сircle С tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    testTail
  </div>
</div>
`;

exports[`Тестирует компонент сircle С буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирует компонент сircle С пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
